core/src/wallet/stellar/explorers/horizon/HorizonApiParser.hpp

```diff
@@ -0,0 +1,228 @@
+/*
+ *
+ * ledger-core
+ *
+ * Created by Pierre Pollastri.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#ifndef HORIZON_API_PARSER_HPP
+#define HORIZON_API_PARSER_HPP
+
+#include <rapidjson/reader.h>
+#include <utils/JsonParserPath.hpp>
+
+namespace ledger {
+    namespace core {
+        struct HorizonError {
+            uint32_t statusCode;
+            std::string statusMessage;
+            std::string errorTitle;
+            std::string errorDetails;
+
+            HorizonError() : statusCode(0) {};
+        };
+
+        template <class ResultType, class Parser, bool HasEmbedded = true>
+        class HorizonApiParser {
+        public:
+            using Result = ResultType;
```

Comment:
In my opinion aliasing template type parameter without any trait operation on it is a bad practice and add useless complexity. Can you consider to just rename `ResultType` to `Result` or at least decay `ResultType` ?